lib/dotcom/system_status/groups.ex

```diff
@@ -0,0 +1,337 @@
+defmodule Dotcom.SystemStatus.Groups do
+  @moduledoc """
+  A module that groups alerts into statuses for the system status widget.
+  """
+
+  alias Alerts.Alert
+
+  # &groups/2 returns an ordered data structure, sorted in the order
+  # given by `@lines`.
+  @lines ["Blue", "Orange", "Red", "Green"]
+  @green_line_branches ["Green-B", "Green-C", "Green-D", "Green-E"]
+  @routes ["Blue", "Mattapan", "Orange", "Red"] ++ @green_line_branches
+
+  def groups(alerts, time) do
+    grouped_alerts = Map.new(@routes, &{&1, alerts_for_line(alerts, &1)})
+
+    @routes
+    |> Enum.map(fn route_id ->
+      statuses =
+        grouped_alerts
```

Comment:
Why not `Enum.map` over the `grouped_alerts`?

Reply:
Heh - great question!

I think I refactored back and forth exactly what was happening just inside versus just outside the `Enum.map` call, and then stopped _just short_ of optimal.